diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7415eab..a3cd312 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -397,7 +397,8 @@ if (WIN32 AND HAVE_SNPRINTF)
 endif (WIN32 AND HAVE_SNPRINTF)
 
 if (gflags_FOUND)
-  target_include_directories (glog PUBLIC $<BUILD_INTERFACE:${gflags_INCLUDE_DIR}>)
+  # when set to PUBLIC, will cause some wrong interface_include_directory for vcpkg
+  target_include_directories (glog PRIVATE $<BUILD_INTERFACE:${gflags_INCLUDE_DIR}>)
   target_link_libraries (glog PUBLIC ${gflags_LIBRARIES})
 
   if (NOT BUILD_SHARED_LIBS)
